src/settings-ui/Settings.UI/Controls/ShortcutControl/ShortcutControl.xaml.cs

```diff
@@ -134,26 +133,50 @@ private void ShortcutControl_Unloaded(object sender, RoutedEventArgs e)
 
         private void KeyEventHandler(int key, bool matchValue, int matchValueCode)
         {
-            switch ((VirtualKey)key)
+            VirtualKey virtualKey = (VirtualKey)key;
+            switch (virtualKey)
             {
                 case VirtualKey.LeftWindows:
                 case VirtualKey.RightWindows:
+                    if (!matchValue && _modifierKeysOnEntering.ContainsKey(virtualKey))
+                    {
+                        SendSingleKeyboardInput((short)virtualKey, (uint)NativeKeyboardHelper.KeyEventF.KeyUp);
+                        _modifierKeysOnEntering.Remove(virtualKey);
+                    }
+
                     internalSettings.Win = matchValue;
                     break;
                 case VirtualKey.Control:
                 case VirtualKey.LeftControl:
                 case VirtualKey.RightControl:
+                    if (!matchValue && _modifierKeysOnEntering.ContainsKey(VirtualKey.Control))
```

Comment:
is that ok that we're switching on `VirtualKey.RightControl` here, even though we always add `VirtualKey.Control`? Same question for other modifiers. Perhaps that logic could be written outside of switch once to be more robust?

Reply:
The `_modifierKeysOnEntering ` should only add `VirtualKey.Control` to indicate that the user is holding down a Control key, as for which Control key is actually pressed and held, we should know that from the `virtualKey ` variable. Thus, in the next line of `SendSingleKeyboardInput` , I will send `virtualKey` instead of always sending `VirtualKey.Control`, I think this should solve the problem.